### PR TITLE
WIP: Remove social login dependency on username & password

### PR DESCRIPTION
**Pull request by shayneczyzewski**
### Description

This removes the dependency of social login on the `usernameAndPassword` auth method fields (`username` and `password`).

Additionally, it is a breaking change since the previously optional `getUserFieldsFn`  is now a required field `getUserFn`. This is expected to return a User entity. This change allows users to either find an existing user or create one based on the information from the social login (i.e. associate a Google user with an existing account with that email).
 
### Select what type of change this PR introduces:

1. [ ] **Just code/docs improvement** (no functional change). 
2. [ ] **Bug fix** (non-breaking change which fixes an issue).
3. [ ] **New feature** (non-breaking change which adds functionality).
4. [x] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

### Update Waspc ChangeLog and version if needed
If you did a bug fix, new feature, or breaking change, that affects waspc, make sure you satisfy the following:
1. [ ] I updated [ChangeLog.md](https://github.com/wasp-lang/wasp/blob/main/waspc/ChangeLog.md) with description of the change this PR introduces.
2. [ ] I bumped waspc version in [waspc.cabal](https://github.com/wasp-lang/wasp/blob/main/waspc/waspc.cabal) to reflect changes I introduced, with regards to the version of the latest wasp release, if the bump was needed.

### CODE REVIEW

File: waspc/data/Generator/templates/server/src/core/auth/prismaMiddleware.js [hunk 1]
```diff
@@ -60,12 +60,13 @@ export const registerAuthMiddleware = (prismaClient) => {
...
 
-const userValidations = [
-  { validates: PASSWORD_FIELD, message: 'password must be present', validator: password => !!password },
-  { validates: PASSWORD_FIELD, message: 'password must be at least 8 characters', validator: password => password.length >= 8 },
-  { validates: PASSWORD_FIELD, message: 'password must contain a number', validator: password => /\d/.test(password) },
-]
+const userValidations = []
 
+{=# isEmailOrUsernameAndPasswordAuthEnabled =}
+userValidations.push({ validates: PASSWORD_FIELD, message: 'password must be present', validator: password => !!password })
```
Comment (infomiho) [inline waspc/data/Generator/templates/server/src/core/auth/prismaMiddleware.js:66]: @shayneczyzewski We need password validation for both username and email & password auth method.

Maybe pass into the template `isPasswordFieldValidated` or similar to encapsulate that logic.

File: waspc/examples/todoApp/src/client/pages/auth/Login.jsx [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import { Link } from 'react-router-dom'
 
 import { LoginForm } from '@wasp/auth/email'
-// import { SignInButton as GoogleSignInButton } from '@wasp/auth/helpers/Google'
```
Comment (infomiho) [inline waspc/examples/todoApp/src/client/pages/auth/Login.jsx:4]: I would remove these changes 👍

File: waspc/data/Generator/templates/server/src/core/auth/prismaMiddleware.js (hunk 1 above)
Comment (infomiho) [reply waspc/data/Generator/templates/server/src/core/auth/prismaMiddleware.js:66]: My bad, you are checking both!

I didn't read carefully the variable name `isEmailOrUsernameAndPasswordAuthEnabled` but I would still maybe rename it :)

**Comment by infomiho** [comment]
Should we close this PR in favor of:
- https://github.com/wasp-lang/wasp/pull/1102

**Comment by shayneczyzewski** [comment]
Yes, closing in favor of the simpler version for now @infomiho 👍🏻

**Pull request closed without merging**